### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - master
+
+jobs:
+- job: "Sonar"
+  pool:
+    vmImage: 'windows-latest'
+    demands:
+    - msbuild
+  steps:
+    - task: SonarCloudPrepare@1
```
Comment (jennyf19) [inline azure-pipelines.yml:17]:
```suggestion
    - template: build/template-install-dotnet-core.yaml
    - task: SonarCloudPrepare@1
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: 'restore'
+    - task: MSBuild@1
+      inputs:
+        solution: 'Microsoft.Identity.Web.sln'
+    - task: DotNetCoreCLI@2
+      displayName: 'Run unit tests'
+      inputs:
+        command: test
```
Comment (jennyf19) [inline azure-pipelines.yml:36]:
```suggestion
        solution: 'Microsoft.Identity.Web.sln'
        command: test
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        SonarCloud: 'SonarCloud'
+        organization: 'azuread'
+        scannerMode: 'MSBuild'
+        projectKey: 'AzureAD_microsoft-identity-web'
+        projectName: 'microsoft-identity-web'
+        extraProperties: |
+          sonar.exclusions=tests\**
+    - task: DotNetCoreCLI@2
+      inputs:
+        command: 'restore'
```
Comment (jennyf19) [inline azure-pipelines.yml:29]:
```suggestion
        solution: 'Microsoft.Identity.Web.sln'
        command: 'restore'
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      inputs:
+        solution: 'Microsoft.Identity.Web.sln'
+        command: 'restore'
+    - task: MSBuild@1
+      inputs:
+        solution: 'Microsoft.Identity.Web.sln'
+    - task: DotNetCoreCLI@2
+      displayName: 'Run unit tests'
+      inputs:
+        solution: 'Microsoft.Identity.Web.sln'
```
Comment (jennyf19) [inline azure-pipelines.yml:37]:
```suggestion
```

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  strategy:
+    matrix:
+      windows:
+        imageName: 'windows-latest'
+  pool:
+    vmImage: $(imageName)
+  steps:
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: 'restore'
```
Comment (jennyf19) [inline azure-pipelines.yml:55]:
```suggestion
      solution: 'Microsoft.Identity.Web.sln'
      command: 'restore'
```

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  pool:
+    vmImage: $(imageName)
+  steps:
+  - task: DotNetCoreCLI@2
+    inputs:
+      solution: 'Microsoft.Identity.Web.sln'
+      command: 'restore'
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: 'build'
```
Comment (jennyf19) [inline azure-pipelines.yml:59]:
```suggestion
      solution: 'Microsoft.Identity.Web.sln'
      command: 'build'
```

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        SonarCloud: 'SonarCloud'
+        organization: 'azuread'
+        scannerMode: 'MSBuild'
+        projectKey: 'AzureAD_microsoft-identity-web'
+        projectName: 'microsoft-identity-web'
+        extraProperties: |
+          sonar.exclusions=tests\**
+    - task: DotNetCoreCLI@2
+      inputs:
+        solution: 'Microsoft.Identity.Web.sln'
```
Comment (jennyf19) [inline azure-pipelines.yml:29]:
```suggestion
        solution: './Microsoft.Identity.Web.sln'
```

File: azure-pipelines.yml [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        SonarCloud: 'SonarCloud'
+        organization: 'azuread'
+        scannerMode: 'MSBuild'
+        projectKey: 'AzureAD_microsoft-identity-web'
+        projectName: 'microsoft-identity-web'
+        extraProperties: |
+          sonar.exclusions=tests\**
+    - task: DotNetCoreCLI@2
+      inputs:
+        solution: './Microsoft.Identity.Web.sln'
```
Comment (jennyf19) [inline azure-pipelines.yml:29]:
```suggestion
        projects: 'Microsoft.Identity.Web.sln'
```

**Review by jmprieur — APPROVED** [review]
LGTM. Thanks @jennyf19 

A few questions: do we want to have that part of PRs (we can try for a few days and see if we like it), or have other triggers: https://identitydivision.visualstudio.com/IDDP/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=1201&view=Tab_Triggers

**Comment by jennyf19** [comment]
idk...what do you want to do? @jmprieur

**Pull request merged**